### PR TITLE
Use custom header for login

### PR DESCRIPTION
**Pull request by deanobarnett**
## Background
As a Zuora user, I would like my custom headers to be included in all requests. 

## Why:

* Customer headers are not being used for login action

## This change addresses the need by:

* Adding the customer headers to the login soap request
* cleaning up some random spec failures caused by old object staying around. 
* Using before/after all to speed up integration spec a bit.

### CODE REVIEW

**Comment by deanobarnett** [comment]
@anfleene Hey. Would you mind casting your eye over this? We are running into some issues with passing custom headers and I think this should resolve it. Cheers!

File: spec/subscribe_integration_spec.rb [hunk 1]
```diff
@@ -38,11 +36,7 @@
         )
     end
 
-    after do
-      @product.delete
-      @account.delete
-      @account_2.delete if !!@account_2
```
Comment (anfleene) [inline spec/subscribe_integration_spec.rb]: Wont this leave test accounts around after the tests run?

**Comment by anfleene** [comment]
@deanobarnett sorry for being slow on this. 

This looks good thanks for the addition.

In addition to my question about the test changes, can you add a section to the README on how to set custom headers?

Thanks!

File: spec/subscribe_integration_spec.rb (hunk 1 above)
Comment (deanobarnett) [inline spec/subscribe_integration_spec.rb]: Good shout, will update this

**Comment by deanobarnett** [comment]
@anfleene :+1: Updates have been applied

**Comment by anfleene** [comment]
Awesome thanks. I'll cut a release shortly.

**Comment by anfleene** [comment]
active_zuora (2.3.0) is out on rubygems

**Comment by deanobarnett** [comment]
@anfleene Thank you kindly!

**Pull request merged**